### PR TITLE
Custom draw controls & better instructions

### LINKED ISSUE #248: Restore ability to edit zoning district labels

### LINKED ISSUE #253: Add polygon icon to instructions

### PR DESCRIPTION
**Pull request by andycochran**
This PR… 
- Adds custom draw controls using FontAwesome icons
- Improves the language of the draw instructions
- Adds inline icons referring to the draw controls to the draw instructions
- Hides the regular draw controls * 

Closes #253 

- Restores the regressed feature editing zoning geom labels 

Closes #248 

-----

_* This is a big of a hack. You can't set `controls` to null without preventing the Delete/Backspace keys from working with draw. So CSS is hiding the 2nd top-left control, which is coincidentally the draw controls. This could effect other map controls that are selected by `.mapboxgl-ctrl-top-left > .mapboxgl-ctrl:nth-child(2)`._

### CODE REVIEW

File: app/components/project-geometries/draw-lots-to-union.js [hunk 1]
```diff
@@ -60,11 +60,19 @@ export default class DrawLotsToUnion extends Component {
 
       const drawStateCallback = () => {
         this.set('currentDrawing', draw.getAll());
+        this.set('drawMode', draw.getMode());
+
+        const selectedFeatures = draw.getSelected().features;
```
Comment (allthesignals) [inline app/components/project-geometries/draw-lots-to-union.js]:
```suggestion
        const { features: selectedFeatures } = draw.getSelected();
```

File: app/templates/components/project-geometries/draw-lots-to-union.hbs [hunk 2]
```diff
@@ -21,11 +21,30 @@
   {{/map.source}}
 {{/if}}
 
+<div class="draw-controls">
+  <button class="polygon {{if (eq drawMode 'draw_polygon') 'active'}}" {{action 'handleDrawButtonClick'}}>{{fa-icon 'draw-polygon' size='2x' fixedWidth=true}}</button>
+  <button class="trash" {{action 'handleTrashButtonClick'}}>{{fa-icon 'trash-alt' prefix='far' size='2x' fixedWidth=true}}</button>
+</div>
```
Comment (allthesignals) [inline app/templates/components/project-geometries/draw-lots-to-union.hbs]: What do you think about moving this to a separate component? We could pass it `map` for the mode state, `draw` for the mode state, and a closure action that is fired when `modechange` occurs.

File: app/components/project-geometries/draw-lots-to-union.js [hunk 3]
```diff
@@ -60,11 +60,19 @@ export default class DrawLotsToUnion extends Component {
...
+        if (selectedFeatures.length === 1) {
+          this.set('selectedZoningFeature', selectedFeatures[0]);
+        } else {
+          this.set('selectedZoningFeature', undefined);
+        }
```
Comment (allthesignals) [inline app/components/project-geometries/draw-lots-to-union.js]: Zoning is specific to underlying zoning and two other things - `draw-lots-to-union` is supposed to generic. This component could keep track of "selected" (clicked, but not necessarily drawn?)  features (as you have in line 65) but should have nothing to do with "domain" concerns like zoning, development site, etc.

File: app/components/project-geometries/draw-lots-to-union.js [hunk 4]
```diff
@@ -80,6 +88,9 @@ export default class DrawLotsToUnion extends Component {
   @argument
   mode;
 
+  @argument
+  editableLabel = false;
```
Comment (allthesignals) [inline app/components/project-geometries/draw-lots-to-union.js]: I wonder if we could _compose_ this type of thing by splitting it out into its own component and injecting it where it's needed (zoning-related geometries)

File: app/templates/components/project-geometries/draw-lots-to-union.hbs (hunk 2 above)
Comment (andycochran) [reply app/templates/components/project-geometries/draw-lots-to-union.hbs]: If you think it's necessary. Is it really bloating the `draw-lots-to-union` all that much? What's the advantage of separating it if it won't be used in other places?

File: app/components/project-geometries/draw-lots-to-union.js (hunk 3 above)
Comment (andycochran) [reply app/components/project-geometries/draw-lots-to-union.js]: I've changed `selectedZoningFeature` to `selectedFeature`.

File: app/components/project-geometries/draw-lots-to-union.js (hunk 4 above)
Comment (andycochran) [reply app/components/project-geometries/draw-lots-to-union.js]: Isn't that more complicated that just setting the boolean for the 3 geom types that need it?

File: app/templates/components/project-geometries/draw-lots-to-union.hbs (hunk 2 above)
Comment (allthesignals) [reply app/templates/components/project-geometries/draw-lots-to-union.hbs]: I don't think _reuse_ is the main benefit of separating out concerns - it's just good practice, easier to reason about, and makes things easier to test out.

**Pull request merged**